js/miniQuery.js

```diff
@@ -0,0 +1,157 @@
+/*!
+ * minQuery
+ */
+"use strict"
+
+ var SweetSelector = {};
+
+ SweetSelector.select = function (cssSelector){
+  switch (cssSelector.charAt(0)) {
+  case '.':
+    return document.getElementsByClassName(cssSelector.slice(1));
+  case "#":
+    return document.getElementById(cssSelector.slice(1));
+  default:
+    return document.getElementsByTagName(cssSelector);
+  }
+ }
+
+var MiniQuery = function (cssSelector) {
+  if(typeof(cssSelector) == "function") {
+    return $.ready(cssSelector) }
+  else {
+    return SweetSelector.select(cssSelector)}
+};
+
+var $ = MiniQuery;
+
+MiniQuery.ready = function(thing){document.addEventListener('DOMContentLoaded', function(event){thing()})};
+
+HTMLCollection.prototype.hide = function() {
```

Comment:
using the prototype here can be dangerous. you only want selectors that your miniquery returns to have these properties, not all `HTMLCollection` objects. You don't want someone who includes your library to suddenly be able to call new methods on standard JS objects.